pkg/segment/structs/segstructs_test.go

```diff
@@ -126,3 +126,79 @@ func Test_HasQueryAggergatorBlock(t *testing.T) {
 
 	assert.Equal(t, true, qa.HasQueryAggergatorBlock(), "Expected true when HasQueryAggergatorBlock is true, got false")
 }
+
+func Test_HasQueryAggergatorBlockInChain_EmptyChain(t *testing.T) {
+	qa := &QueryAggregators{}
+
+	assert.False(t, qa.HasQueryAggergatorBlockInChain(), "Expected false when the chain is empty, got true")
+}
+
+func Test_HasQueryAggergatorBlockInChain_SingleNodeWithBlock(t *testing.T) {
+	qa := &QueryAggregators{
+		OutputTransforms: &OutputTransforms{
+			MaxRows:   10,
+			RowsAdded: 1,
+		},
+	}
+
+	assert.True(t, qa.HasQueryAggergatorBlockInChain(), "Expected true when single node has a query aggregator block, got false")
+}
+
+func Test_HasQueryAggergatorBlockInChain_SingleNodeWithoutBlock(t *testing.T) {
+	qa := &QueryAggregators{
+		OutputTransforms: &OutputTransforms{
+			MaxRows:   1,
+			RowsAdded: 10,
+		},
+	}
+
+	assert.False(t, qa.HasQueryAggergatorBlockInChain(), "Expected false when single node does not have a query aggregator block, got true")
+}
+
+func Test_HasQueryAggergatorBlockInChain_MultipleNodesWithBlockAtEnd(t *testing.T) {
+	qa := &QueryAggregators{
+		Next: &QueryAggregators{
+			OutputTransforms: &OutputTransforms{
+				MaxRows:   10,
+				RowsAdded: 1,
+			},
+		},
+	}
+
+	assert.True(t, qa.HasQueryAggergatorBlockInChain(), "Expected true when a node in the chain has a query aggregator block, got false")
+}
+
+func Test_HasQueryAggergatorBlockInChain_MultipleNodesWithoutBlock(t *testing.T) {
+	qa := &QueryAggregators{
+		Next: &QueryAggregators{},
+	}
+
+	assert.False(t, qa.HasQueryAggergatorBlockInChain(), "Expected false when no nodes in the chain have a query aggregator block, got true")
+}
+
+func Test_HasQueryAggergatorBlockInChain_MultipleNodesWithBlockAtStart(t *testing.T) {
+	qa := &QueryAggregators{
+		OutputTransforms: &OutputTransforms{
+			MaxRows:   10,
+			RowsAdded: 1,
+		},
+		Next: &QueryAggregators{},
+	}
+
+	assert.True(t, qa.HasQueryAggergatorBlockInChain(), "Expected true when the first node in the chain has a query aggregator block, got false")
+}
+
+func Test_HasQueryAggergatorBlockInChain_MultipleNodesWithBlockInMiddle(t *testing.T) {
+	qa := &QueryAggregators{
+		Next: &QueryAggregators{
+			Next: &QueryAggregators{
+				OutputTransforms: &OutputTransforms{
+					MaxRows:   10,
+					RowsAdded: 1,
+				},
+			},
+		},
+	}
+
+	assert.True(t, qa.HasQueryAggergatorBlockInChain(), "Expected true when a middle node in the chain has a query aggregator block, got false")
+}
```

Comment:
nit: this test doesn't quite match it's name since the block causing it to be true is at the end of the chain. If you add another empty QueryAggregators to the end of the chain, this test will be good

Reply:
Thank you, I've fixed it.